pkg/release/release.go

```diff
@@ -108,29 +171,81 @@ func IsDirtyBuild(build string) bool {
 	return strings.Contains(build, "dirty")
 }
 
+// TODO: Consider collapsing some of these functions.
+//       Keeping them as-is for now as kubepkg is dependent on them.
+func GetStableReleaseKubeVersion() (string, error) {
+	logrus.Info("Retrieving Kubernetes release version...")
+	return GetKubeVersion("https://dl.k8s.io/release/stable.txt")
+}
+
+func GetStablePrereleaseKubeVersion() (string, error) {
+	logrus.Info("Retrieving Kubernetes testing version...")
+	return GetKubeVersion("https://dl.k8s.io/release/latest.txt")
+}
+
+func GetLatestCIKubeVersion() (string, error) {
+	logrus.Info("Retrieving Kubernetes latest build version...")
+	return GetKubeVersion("https://dl.k8s.io/ci/latest.txt")
+}
+
+func GetCIKubeVersion(branch string) (string, error) {
+	logrus.Infof("Retrieving Kubernetes build version on the '%s' branch...", branch)
+	// TODO: We may need to check if the branch exists first to handle the branch cut scenario
+	versionMarker := "latest"
+	if branch != "master" {
+		version := strings.TrimPrefix(branch, "release-")
+
+		versionMarker = fmt.Sprintf("%s-%s", versionMarker, version)
+	}
+
+	versionMarkerFile := fmt.Sprintf("%s.txt", versionMarker)
+	logrus.Infof("Version marker file: %s", versionMarkerFile)
+
+	u, parseErr := url.Parse("https://dl.k8s.io/ci")
```

Comment:
The same comment with putting the url at the top of the file applies here too.

Reply:
@bartsmykla -- Hehe. I was actually planning on doing that for a follow-up.
I want to think a little more about version markers before parameterizing them.